x-pack/plugins/cases/public/components/app/routes.tsx

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback } from 'react';
+import { Redirect, Route, Switch } from 'react-router-dom';
+import { AllCases } from '../all_cases';
+import { CaseView } from '../case_view';
+import { CreateCase } from '../create';
+import { ConfigureCases } from '../configure_cases';
+import { CasesRoutesProps } from './types';
+import { useCasesContext } from '../cases_context/use_cases_context';
+import {
+  getCasesConfigurePath,
+  getCreateCasePath,
+  getCaseViewPath,
+  getCaseViewWithCommentPath,
+  getSubCaseViewPath,
+  getSubCaseViewWithCommentPath,
+  useAllCasesNavigation,
+  useCaseViewNavigation,
+} from '../../common/navigation';
+import { NoPrivilegesPage } from '../no_privileges';
+import * as i18n from './translations';
+
+const CasesRoutesComponent: React.FC<CasesRoutesProps> = (props) => {
+  const { timelineIntegration } = props;
+  const { basePath, userCanCrud } = useCasesContext();
+  const { navigateToAllCases } = useAllCasesNavigation();
+  const { navigateToCaseView } = useCaseViewNavigation();
+  
+  const onCreateCaseSuccess = useCallback(
+    async ({ id }) => navigateToCaseView({ detailName: id }),
+    [navigateToCaseView]
+  );
+
+  return (
+    <Switch>
+      <Route strict exact path={basePath}>
+        <AllCases {...props} />
```

Comment:
Using the spreading operator can cause unnecessary re-renders because we pass unnecessary props to the component. I think it is better to explicitly set the props.

Reply:
yes this is temporary, I will set up all the props individually